### PR TITLE
Use sync 1.4.0

### PR DESCRIPTION
**Pull request by radu-tutueanu**

### CODE REVIEW

**Comment by radu-tutueanu** [comment]
@fealebenpae All the react tests fail. Is that a known issue?

File: src/js_realm.hpp [hunk 1]
```diff
@@ -44,6 +44,20 @@
 namespace realm {
 namespace js {
 
+static std::string normalize_realm_path(std::string path) {
```
Comment (kristiandupont) [inline src/js_realm.hpp]: Shouldn't this live in object store now, if we are sharing it?

Comment (radu-tutueanu) [inline src/js_realm.hpp]: I think you are right, it does make sense to move it to ObjectStore

Comment (kristiandupont) [inline src/js_realm.hpp]: But for this release we can go with this I suppose.

**Pull request merged**